go-controller/pkg/node/gateway_udn.go

```diff
@@ -167,6 +179,32 @@ func (udng *UserDefinedNetworkGateway) AddNetwork() error {
 	if err != nil {
 		return fmt.Errorf("could not add VRF %d for network %s, err: %v", vrfTableId, udng.GetNetworkName(), err)
 	}
+	if udng.openflowManager != nil {
+		udng.openflowManager.addNetwork(udng.NetInfo, udng.masqCTMark)
+
+		waiter := newStartupWaiter()
+		readyFunc := func() (bool, error) {
+			if err := setBridgeNetworkOfPorts(udng.openflowManager.defaultBridge, udng.GetNetworkName()); err != nil {
+				return false, fmt.Errorf("failed to set network %s's openflow ports for default bridge; error: %v", udng.GetNetworkName(), err)
+			}
+			if udng.openflowManager.externalGatewayBridge != nil {
+				if err := setBridgeNetworkOfPorts(udng.openflowManager.externalGatewayBridge, udng.GetNetworkName()); err != nil {
+					return false, fmt.Errorf("failed to set network %s's openflow ports for secondary bridge; error: %v", udng.GetNetworkName(), err)
+				}
+			}
+			return true, nil
+		}
+		postFunc := func() error {
+			if err := udng.Reconcile(); err != nil {
+				return fmt.Errorf("failed to reconcile flows on bridge for network %s; error: %v", udng.GetNetworkName(), err)
+			}
+			return nil
+		}
+		waiter.AddWait(readyFunc, postFunc)
+		if err := waiter.Wait(); err != nil {
```

Comment:
so using wait we could wait up to 300 seconds and block the network manager from creating any other networks (cause threadiness is 1).

Reply:
maybe? I guess that's something we can tune when we do perf/scale testing. It really depends on how many networks we expect users to create at a time and how long it takes to create those networks. But do we really expect it to take 300 seconds for ovn-controller to create the patch port? I think we used 300 seconds as our waiter for start up when we used centralized OVN and it could take a long time on a large scale cluster for everything to reconcile. This is during runtime when things should be mostly in a steady state, so could we lower the maximum wait timer? I would especially think this would be much faster in IC.

Reply:
No we definitely don't expect ovn to take so long to create the patch ports.  In theory it shouldn't be longer than the time it takes to provision any pod port.  So max "a few seconds" at scale?

Reply:
OK, I changed the waiter to accept a custom timeout and used 30 seconds here.